### PR TITLE
D3ASIM-3553: Rename Inter Area Agent to Market Agent

### PR DESCRIPTION
**Pull request by fievelk**
Please review this through commits. The relevant changes are just a bunch of lines, while the rest is linting and imports fixes.

### CODE REVIEW

File: gsy_framework/constants_limits.py [hunk 1]
```diff
@@ -187,7 +211,16 @@ class AlternativePricing:
             FEED_IN_TARIFF_PERCENTAGE = 50
             ALT_PRICING_MARKET_MAKER_NAME = "AGENT"
 
+    class IAASettings(MASettings):
```
Comment (fievelk) [inline gsy_framework/constants_limits.py:214]: Here we keep the old `IAASettings` class for backwards compatibility.

File: gsy_framework/sim_results/__init__.py [hunk 2]
```diff
@@ -15,77 +15,91 @@
...
+        area_name_from_area_or_ma_name(trade["buyer"]) == area_name
+        and area_name_from_area_or_ma_name(trade["seller"]) in child_names)
 
 
 def is_trade_external(trade, area_name, child_names):
-    return area_sells_to_child(trade, area_name, child_names) or \
-        child_buys_from_area(trade, area_name, child_names)
-
-
-def area_name_from_area_or_iaa_name(name):
```
Comment (fievelk) [inline gsy_framework/sim_results/__init__.py:90]: This method was duplicate. I removed it and now we use the one defined in `utils.py`

File: gsy_framework/utils.py [hunk 3]
```diff
@@ -306,51 +333,51 @@ def get_area_uuid_name_mapping(area_dict, results):
...
-    else:
-        dict[key] = 0 - value
-    return dict
 
-
-def make_iaa_name_from_dict(owner):
-    return f"IAA {owner['name']}"
+def make_ma_name_from_dict(owner: Dict) -> str:
+    """Generate the name of the marget agent from the given owner."""
+    return f"MA {owner['name']}"
```
Comment (fievelk) [inline gsy_framework/utils.py:365]: This change affects many parts of the code across repos. This is why I needed to rename IAA strings everywhere.

File: gsy_framework/utils.py [hunk 4]
```diff
@@ -389,53 +416,68 @@ def deep_size_of(input_obj):
...
 
 
-def area_name_from_area_or_iaa_name(name):
-    return name[4:] if name[:4] == 'IAA ' else name
+def area_name_from_area_or_ma_name(name: str) -> str:
+    """Retrieve the name of the area.
+
+    The input can be either the area name itself or the name of the market agent of the area.
+    """
+    return name[3:] if name[:3] == "MA " else name
```
Comment (fievelk) [inline gsy_framework/utils.py:434]: Finally: since `MA` is shorter than `IAA`, this check now uses 3 characters rather than 4.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]

**Comment by fievelk** [comment]
> @fievelk is there something missing from this feature? I see that it's not merged yet

@Hassan754 it's not merged because unrelated `master` tests are failing for gsy-e-sdk. We decided to always test everything before merging, but I saw that other PRs got merged in the meanwhile so I'll proceed with this as well.

**Pull request merged**